### PR TITLE
Methods of RegExp and String

### PR DESCRIPTION
**Pull request by fsdude**
Translation to portuguese done. Followed the english version again.

Source file can be found: https://github.com/javascript-tutorial/en.javascript.info/blob/master/9-regular-expressions/17-regexp-methods/article.md

### CODE REVIEW

**Comment by odsantos** [comment]
Hi @fsdude  would you please resolve the conflicts, or kindly inform if unavailable to? Thank you.

**Comment by fsdude** [comment]
> Hi @fsdude would you please resolve the conflicts, or kindly inform if unavailable to? Thank you.

Hello! Yeah.. lately I've been busy with some work. I don't know if I'll be able to fix this soon, sorry for that =)
Thanks by the way

**Review by odsantos — CHANGES_REQUESTED** [review]
Hi @kiniggit, @luizclr, @adriavieira314,  @Mr-Saxobeat, please since @fsdude might be unavailable I would appreciate if any of you could review this PR #141 , including my revision.
Thank you.

**Comment by odsantos** [comment]
Hi @kiniggit, @luizclr, @adriavieira314, @Mr-Saxobeat, please since @fsdude might be unavailable I would appreciate if any of you could review this PR #141 , including my changes.
Thank you.

**Comment by Peruibeloko** [comment]
@odsantos aqui também tem o problema da tradução dos `pattern:`

**Comment by odsantos** [comment]
> @odsantos aqui também tem o problema da tradução dos `pattern:`

@Peruibeloko I'm afraid I do not know what you intend to say with this phrase. Please, could you share more details? Thanks.

**Comment by odsantos** [comment]
> @odsantos aqui também tem o problema da tradução dos `pattern:`

@Peruibeloko I'm afraid I do not know what you intend to say with this phrase. Please, could you share more details? Thanks.

**Comment by Peruibeloko** [comment]
@odsantos the PRs #147, #141 (this one) and #274 all translated parts of the text that weren't meant to have been translated.

The parts in question are the \`pattern:\`, \`subject:\` and \`match:\` decorators used in the Regex chapters, as they're used by the site builder literally. Changing this might break the builds! My evidence for this is that all of them match the class names used in the HTML output of the site.

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 1]
```diff
@@ -1,198 +1,198 @@
...
 ## str.match(regexp)
 
-The method `str.match(regexp)` finds matches for `regexp` in the string `str`.
+O método `str.macth(regexp)` pesquisa correspondências para `regexp` na string `str`.
 
-It has 3 modes:
+Possui 3 modos:
 
-1. If the `regexp` doesn't have flag `pattern:g`, then it returns the first match as an array with capturing groups and properties `index` (position of the match), `input` (input string, equals `str`):
+1. Se a `regexp` não tiver a flag `padrão:g`, ela retornará a primeira correspondência como uma matriz com a captura de grupos e propriedades `index` (posição da correspondência), `input` (string de entrada, igual a `str`):
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:11]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 2]
```diff
@@ -1,198 +1,198 @@
...
-    // Additional information:
-    alert( result.index );  // 7 (match position)
-    alert( result.input );  // I love JavaScript (source string)
+    // Informação adicional:
+    alert( result.index );  // 7 (posição da correspondÊncia)
+    alert( result.input );  // Eu amo JavaScript (string original)
     ```
 
-2. If the `regexp` has flag `pattern:g`, then it returns an array of all matches as strings, without capturing groups and other details.
+2. Se a `regexp` tiver a flag `padrão:g`, ela retornará uma matriz com todas as correspondências como strings, sem capturar grupos e outros detalhes.
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:27]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 3]
```diff
@@ -1,198 +1,198 @@
...
+    let str = "Eu amo JavaScript";
 
     let result = str.match(/Java(Script)/g);
 
     alert( result[0] ); // JavaScript
     alert( result.length ); // 1
     ```
 
-3. If there are no matches, no matter if there's flag `pattern:g` or not, `null` is returned.
+3. Se não houver correspondências, não importa se há flag `padrão:g` ou não, `null` é retornado.
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:37]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 4]
```diff
@@ -1,198 +1,198 @@
...
+    Se desejamos que o resultado seja uma matriz, podemos escrever assim:
 
     ```js
     let result = str.match(regexp) || [];
     ```
 
 ## str.matchAll(regexp)
 
-[recent browser="new"]
+[navegador recente="new"]
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:58]: This shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 5]
```diff
@@ -1,198 +1,198 @@
...
+No entanto há uma armadilha.
 
-**When the first argument of `replace` is a string, it only replaces the first match.**
+**Quando o primeiro argumento do `replace` é uma string, ele apenas substitui a primeira correspondência**
 
-You can see that in the example above: only the first `"-"` is replaced by `":"`.
+Você pode ver isso no exemplo acima: apenas o primeiro `"-"` é substituído por `":"`.
 
-To find all hyphens, we need to use not the string `"-"`, but a regexp `pattern:/-/g`, with the obligatory `pattern:g` flag:
+Para encontrar todos os hífens, não precisamos usar a string `"-"`, mas a regexp `padrão:/-/g`, com a flag obrigatória `padrão:g`:
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:138]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 6]
```diff
@@ -226,136 +226,119 @@ let result = str.replace(/(?<name>\w+) (?<surname>\w+)/, (...match) => {
...
 ## regexp.exec(str)
 
-The `regexp.exec(str)` method returns a match for `regexp` in the string `str`.  Unlike previous methods, it's called on a regexp, not on a string.
+O método `regexp.exec(str)` retorna uma correspondência para `regexp` na string `str`. Ao contrário dos métodos anteriores, é chamado numa regexp, não em uma string.
 
-It behaves differently depending on whether the regexp has flag `pattern:g`.
+Se comporta de modo diferente dependendo se a regexp possui a flag `padrão:g`.
 
-If there's no `pattern:g`, then `regexp.exec(str)` returns the first match exactly as  `str.match(regexp)`. This behavior doesn't bring anything new.
+Se não houver `padrão:g`, a `regexp.exec(str)` retorna a primeira correspondência exatamente como `str.match(regexp)`. Esse comportamento não traz nada de novo.
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:237]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 7]
```diff
@@ -226,136 +226,119 @@ let result = str.replace(/(?<name>\w+) (?<surname>\w+)/, (...match) => {
...
 
-But if there's flag `pattern:g`, then:
-- A call to `regexp.exec(str)` returns the first match and saves the position immediately after it in the property `regexp.lastIndex`.
-- The next such call starts the search from position `regexp.lastIndex`, returns the next match and saves the position after it in `regexp.lastIndex`.
-- ...And so on.
-- If there are no matches, `regexp.exec` returns `null` and resets `regexp.lastIndex` to `0`.
+Mas se houver a flag `padrão:g`, então:
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:239]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 8]
```diff
@@ -226,136 +226,119 @@ let result = str.replace(/(?<name>\w+) (?<surname>\w+)/, (...match) => {
...
-regexp.lastIndex = 5; // search from 5th position (from the comma)
+let regexp = /\w+/g; // sem a flag "g", propriedade lastIndex é ignorada
+regexp.lastIndex = 3; // pesquisa a partir da 5º posição (a partir da vírgula)
 
-alert( regexp.exec(str) ); // world
+alert( regexp.exec(str) ); // mundo
 ```
 
-If the regexp has flag `pattern:y`, then the search will be performed exactly at the  position `regexp.lastIndex`, not any further.
+Se o regexp tiver flag `padrão:y`, então a pesquisa será realizada exatamente na posição `regexp.lastIndex`, nada mais.
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:277]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 9]
```diff
@@ -226,136 +226,119 @@ let result = str.replace(/(?<name>\w+) (?<surname>\w+)/, (...match) => {
...
 
-Let's replace flag `pattern:g` with `pattern:y` in the example above. There will be no matches, as there's no word at position `5`:
+Vamos substituir a flag `padrão:g` pela `padrão:y` no exemplo acima. Não haverá correspondências, como não haverá palavra na posição `3`:
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:279]: The `pattern:` syntax shouldn't be translated

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 10]
```diff
@@ -226,136 +226,119 @@ let result = str.replace(/(?<name>\w+) (?<surname>\w+)/, (...match) => {
...
 let str = "Bla-bla-bla";
 
-alert( *!*/love/i*/!*.test(str) ); // false
-alert( str.search(*!*/love/i*/!*) != -1 ); // false
+alert( /amo/i.test(str) ); // false
+alert( str.search(/love/i) != -1 ); // false
 ```
 
-If the regexp has flag `pattern:g`, then `regexp.test` looks from `regexp.lastIndex` property and updates this property, just like `regexp.exec`.
+Se a regexp tiver a flag `padrão:g`, então `regexp.test` procura a partir da propriedade `regexp.lastIndex` e atualiza esta propriedade, assim como `regexp.exec`.
```
Comment (Peruibeloko) [inline 9-regular-expressions/17-regexp-methods/article.md:315]: The `pattern:` syntax shouldn't be translated

**Comment by odsantos** [comment]
> @odsantos the PRs #147, #141 (this one) and #274 all translated parts of the text that weren't meant to have been translated.
> 
> The parts in question are the `pattern:`, `subject:` and `match:` decorators used in the Regex chapters, as they're used by the site builder literally. Changing this might break the builds! My evidence for this is that all of them match the class names used in the HTML output of the site.

@Peruibeloko I'll include your corrections as soon as possible. Thank you.

**Comment by nazarepiedady** [comment]
@odsantos, @jonnathan-ls, @Peruibeloko, what do you recommend colleagues to be done in this pull request?

**Review by jonnathan-ls — APPROVED** [review]
@nazarepiedady @odsantos @Peruibeloko 

I updated the PR with the master branch, fixed merge conflicts, translated pending content and applied the suggestions highlighted as code review https://github.com/javascript-tutorial/pt.javascript.info/pull/141/commits/3565c179819894d7389c11ee4d346341f7477164

Can you evaluate this PR again?

I'm also leaving my LGTM as a review

File: 9-regular-expressions/17-regexp-methods/article.md [hunk 11]
```diff
@@ -1,53 +1,53 @@
-# Methods of RegExp and String
+# Métodos de RegExp e String
 
-In this article we'll cover various methods that work with regexps in-depth.
+Neste artigo, abordaremos em profundidade vários métodos que funcionam com regexps.
 
 ## str.match(regexp)
 
-The method `str.match(regexp)` finds matches for `regexp` in the string `str`.
+O método `str.macth(regexp)` pesquisa correspondências para `regexp` na string `str`.
```
Comment (nazarepiedady) [inline 9-regular-expressions/17-regexp-methods/article.md:7]: @jonnathan-ls, thanks for saving some of your time to work on this. It is very praiseworthy.

I would like to point out a little spelling mistake where it is written `str.macth` should be `str.match`.

**Review by nazarepiedady — COMMENTED** [review]
@jonnathan-ls, is all pending work solved?

File: 9-regular-expressions/17-regexp-methods/article.md (hunk 11 above)
Comment (jonnathan-ls) [reply 9-regular-expressions/17-regexp-methods/article.md]: Thanks for the observation, I added the correction

**Review by nazarepiedady — APPROVED** [review]
It was an excellent work, let's go forward.

**Comment by nazarepiedady** [comment]
@jonnathan-ls, you need to approve and merge if you have time for that, too.

**Comment by jonnathan-ls** [comment]
> you need to approve and merge if you have time for that, too.

I left my approval, but I don't see any point in merging without the other reviewers who registered observations not reviewing it again yet.

I don't think it's worth closing change requests from other code reviewers

**Comment by nazarepiedady** [comment]
@jonnathan-ls, we will wait most likely for centuries if we wait for others even when we have two approvals. The current pull request is 4 years old. It is too much time.

**Comment by nazarepiedady** [comment]
@jonnathan-ls, let's do this way. You approve, and we will wait at least one week to get one more approval to merge the changes.

**Comment by nazarepiedady** [comment]
@Peruibeloko, could you help us review this 4 years old pull request?

**Review by Peruibeloko — APPROVED** [review]

**Comment by Peruibeloko** [comment]
LGTM!

**Pull request merged**